buffer/src/main/java/io/netty/buffer/ByteBufUtil.java

```diff
@@ -389,6 +389,89 @@ private static long compareUintBigEndianB(
         return 0;
     }
 
+    private static final class SWARByteSearch {
+
+        private static final long[] SEARCH_PATTERNS;
+
+        static {
+            SEARCH_PATTERNS = new long[256];
+            for (int i = 0; i < SEARCH_PATTERNS.length; ++i) {
+                final byte b = (byte) (i - 128);
+                SEARCH_PATTERNS[i] = compilePattern(b);
+            }
+        }
+
+        private static long compilePattern(byte byteToFind) {
+            long pattern = byteToFind & 0xFFL;
+            return pattern
+                   | (pattern << 8)
+                   | (pattern << 16)
+                   | (pattern << 24)
+                   | (pattern << 32)
+                   | (pattern << 40)
+                   | (pattern << 48)
+                   | (pattern << 56);
+        }
+
+        private static long patternOf(byte b) {
+            return SEARCH_PATTERNS[b + 128];
+        }
+
+        private static int firstAnyPattern(long word, long pattern, boolean leading) {
+            long input = word ^ pattern;
+            long tmp = (input & 0x7F7F7F7F7F7F7F7FL) + 0x7F7F7F7F7F7F7F7FL;
+            tmp = ~(tmp | input | 0x7F7F7F7F7F7F7F7FL);
+            final int binaryPosition = leading? Long.numberOfLeadingZeros(tmp) : Long.numberOfTrailingZeros(tmp);
+            return binaryPosition >>> 3;
+        }
+
+        // TODO: save using any pattern for the 0 search case, if we care
+        private static int firstZeroByte(long word, long pattern, boolean leading) {
+            long tmp = (word & 0x7F7F7F7F7F7F7F7FL) + 0x7F7F7F7F7F7F7F7FL;
+            tmp = ~(tmp | word | 0x7F7F7F7F7F7F7F7FL);
+            final int binaryPosition = leading? Long.numberOfLeadingZeros(tmp) : Long.numberOfTrailingZeros(tmp);
```

Comment:
Since you're doing a little endian check elsewhere, I would consider making little and big endian variants of this method, just because it makes it harder to mix and match

Reply:
Currently this one isn't used yet, but `firstAnyPattern` instead, but I got the good idea, I will do it for both, :+1:

Reply:
Will do it right after breakfast :P